Services/GEV/Utils/classes/class.gevUserUtils.php

```diff
@@ -2192,4 +2210,95 @@ public function seeBiproSuperior() {
 		
 		return $this->hasRoleIn($roles);
 	}
+
+	public function wbdCheckForNew() {
+		return $this->hasDoneWBDRegistration() && $this->hasWBDRelevantRole() && $this->userExists() && $this->isActive() && !$this->hasSpecialUserId()
+				&& $this->entryDatePassed() && $this->isWBDBWVIdEmpty() && !$this->hasWBDType(self::WBD_TP_SERVICE);
+	}
+
+	public function wbdCheckForAffiliate() {
+		$wbd_errors = array(WBD_ERROR_WRONG_USERDATA
+							, WBD_ERROR_USER_SERVICETYPE
+							, WBD_ERROR_USER_DIFFERENT_TP
+							, WBD_ERROR_USER_UNKNOWN
+							, WBD_ERROR_USER_DEACTIVATED);
+
+		return $this->hasDoneWBDRegistration() && $this->hasWBDRelevantRole() && $this->userExists() && $this->isActive() && !$this->hasSpecialUserId()
+				&& $this->entryDatePassed() && !$this->isWBDBWVIdEmpty() && !$this->hasWBDType(self::WBD_TP_SERVICE) 
+				&& !$this->hasOpenWBDErrors($wbd_errors);
+	}
+
+	public function wbdCheckForRelease() {
+		$wbd_errors = array(WBD_ERROR_WRONG_USERDATA
+							, WBD_ERROR_USER_SERVICETYPE
+							, WBD_ERROR_USER_DIFFERENT_TP
+							, WBD_ERROR_USER_UNKNOWN
+							, WBD_ERROR_USER_DEACTIVATED
+							, WBD_ERROR_NO_RELEASE);
+
+		return $this->userExists() && !$this->hasSpecialUserId()
+				&& $this->isExitDatePassed() && !$this->hasExitDateWBD() && $this->hasWBDType(self::WBD_TP_SERVICE) && !$this->isWBDBWVIdEmpty()
+				&& !$this->hasOpenWBDErrors($wbd_errors);
```

Comment:
Nur über die Reihenfolge wie die Punkte in der WBD.php abgearbeitet werden.
Da kommt die Abgabe zur Zeit als aller letzes.